sandbox/plc4c/api/include/plc4c/subscribe.h

```diff
@@ -0,0 +1,244 @@
+
+/**
+ * Returns the plc4c_connection for a give subscription request
+ * @param subscription_request plc4c_subscription_request
+ * @return plc4c_connection
+ */
+plc4c_connection *plc4c_subscription_request_get_connection(
+    plc4c_subscription_request *subscription_request);
+
+/**
+ * Sets the plc4c_connection for a given subscription request
+ * @param subscription_request plc4c_subscription_request
+ * @param connection plc4c_connection
+ */
+void plc4c_subscription_request_set_connection(
+    plc4c_subscription_request *subscription_request,
+    plc4c_connection *connection);
+
+/**
+ * Adds a cyclic (polling) item to the subscription request.
+ * @param subscription_request the subscription-request object.
+ * @param address address for the resource.
+ * @param polling_interval_in_millis interval in millis in which new data should
+ * be mada available.
+ * @return plc4c_return_code
+ */
+plc4c_return_code plc4c_subscription_request_add_cyclic_item(
+    plc4c_subscription_request *subscription_request, char *address,
+    long polling_interval_in_millis);
+
+/**
+ * Adds a change-of-state item to the subscription request.
+ * @param subscription_request the subscription-request object.
+ * @param address address for the resource.
+ * @return plc4c_return_code
+ */
+plc4c_return_code plc4c_subscription_request_add_change_of_state_item(
+    plc4c_subscription_request *subscription_request, char *address);
+
+/**
+ * Adds a event item to the subscription request.
+ * @param subscription_request the subscription-request object.
+ * @param address address for the resource.
+ * @return plc4c_return_code
+ */
+plc4c_return_code plc4c_subscription_request_add_event_item(
+    plc4c_subscription_request *subscription_request, char *address);
+
+/**
+ * Actually executes the subscription-request.
+ * @param connection connection this subscription-request will be executed on.
+ * @param subscription_request the subscription-request object.
+ * @param subscription_request_execution pointer to a data-structure handling
+ * one execution of the subscription-request.
+ * @return plc4c_return_code
+ */
+plc4c_return_code plc4c_subscription_request_execute(
+    plc4c_subscription_request *subscription_request,
+    plc4c_subscription_request_execution **subscription_request_execution);
+
+/**
+ * Check if new data is available for a given subscription handle.
+ * TODO: The subscription handle must contain a reference to the system-task
+ * associated with it as well as to the subscription request.
+ * @param subscription_handle the subscription handle
+ * @return plc4c_return_code
+ */
+bool plc4c_subscription_check_data_available(void *subscription_handle);
+
+/**
+ * Destroys a given subscription-request execution.
+ *
+ * @param subscription_request_execution the subscription-request execution.
+ */
+void plc4c_subscription_request_execution_destroy(
+    plc4c_subscription_request_execution *subscription_request_execution);
+
```

Comment:
I am not sure you would use two methods here, why not one that returns a boolean or an enum?  When would you call one and not the other?

Reply:
I mean that it is confusing how check_finished_successfully  == false is not the same as check_finished_with_error == true